src/Migrate/deploymentscripts/Initialize-AzMigrateReplicationInfrastructure.ps1

```diff
@@ -0,0 +1,386 @@
+
+            Set-AzKeyVaultManagedStorageSasDefinition -AccountName $LogStorageAcName -VaultName $KeyVaultName -Name $cacheStorageAccountSasSecretName -TemplateUri $lsaat -SasType 'account' -ValidityPeriod ([System.Timespan]::FromDays(30))
+            Set-AzKeyVaultManagedStorageSasDefinition -AccountName $GateWayStorageAcName -VaultName $KeyVaultName -Name $gatewayStorageAccountSasSecretName -TemplateUri $gwyat -SasType 'account' -ValidityPeriod ([System.Timespan]::FromDays(30))
+            
+            # Phase 2
+           
+            # ServiceBusConnectionString
+            $serviceBusConnString = "ServiceBusConnectionString"
+            $serviceBusSecretObject = Get-AzKeyVaultSecret -VaultName $KeyVaultName -Name $serviceBusConnString -ErrorVariable notPresent -ErrorAction SilentlyContinue
+            if($serviceBusSecretObject){
+                Write-Host $serviceBusConnString, " for ", $applianceName, $LogStringSkipping
+            }else{
+                $serviceBusRootKey = Get-AzServiceBusKey -ResourceGroupName $ResourceGroupName -Namespace $ServiceBusNamespace -Name "RootManageSharedAccessKey"
+                $secret = ConvertTo-SecureString -String $serviceBusRootKey.PrimaryConnectionString -AsPlainText -Force
+                Set-AzKeyVaultSecret -VaultName $KeyVaultName -Name $serviceBusConnString -SecretValue $secret
+                Write-Host $LogStringCreated, $serviceBusConnString, " for ", $applianceName
+            }  
+
+            # Policy
+            $policyName = $MigratePrefix + $SiteName + "policy"
+            $existingPolicyObject = Get-AzMigrateReplicationPolicy -PolicyName $policyName -ResourceGroupName $ResourceGroupName -ResourceName $VaultName -ErrorVariable notPresent -ErrorAction SilentlyContinue
+            if(!$existingPolicyObject){
+                $providerSpecificPolicy = [Microsoft.Azure.PowerShell.Cmdlets.Migrate.Models.Api20180110.VMwareCbtPolicyCreationInput]::new()
+                $providerSpecificPolicy.AppConsistentFrequencyInMinute = 240
+                $providerSpecificPolicy.InstanceType = "VMwareCbt"
+                $providerSpecificPolicy.RecoveryPointHistoryInMinute = 4320
+                $providerSpecificPolicy.CrashConsistentFrequencyInMinute = 60
+                $existingPolicyObject = New-AzMigrateReplicationPolicy -PolicyName $policyName -ResourceGroupName $ResourceGroupName -ResourceName $VaultName -ProviderSpecificInput $providerSpecificPolicy
+                Write-Host $LogStringCreated, $policyName
+            }else{
+                Write-Host $policyName, $LogStringSkipping
+            }
+
+            # Policy-container mapping
+            $mappingName = "containermapping"
+            $allFabrics = Get-AzMigrateReplicationFabric -ResourceGroupName $ResourceGroupName -ResourceName $VaultName
+            foreach($fabric in $allFabrics){
+                if($fabric.Name -match $applianceName){
```

Comment:
fabric contains vmwareSiteId/physicalSiteId properties. you have to use them to identify right fabric instead appliance name match which is a hacky approach.

Reply:
PS C:\Users\prsadhu\Documents> $obj =  Get-AzMigrateReplicationFabric -ResourceGroupName azmigratepwshtestasr13072020 -ResourceName AzMigrateTestProjectPWSH02aarsvault
PS C:\Users\prsadhu\Documents> $obj


BcdrState                                 : Valid
CustomDetailInstanceType                  : VMwareV2
EncryptionDetailKekCertExpiryDate         :
EncryptionDetailKekCertThumbprint         :
EncryptionDetailKekState                  : None
FriendlyName                              : AzMigratePWSHTc8d1replicationfabric
Health                                    : Normal
HealthErrorDetail                         : {}
Id                                        : /Subscriptions/7c943c1b-5122-4097-90c8-861411bdd574/resourceGroups/azmigratepwshtestas
                                            r13072020/providers/Microsoft.RecoveryServices/vaults/AzMigrateTestProjectPWSH02aarsva
                                            ult/replicationFabrics/AzMigratePWSHTc8d1replicationfabric
InternalIdentifier                        : 501ff8f2-c9d7-5bf4-87ff-d0b3fc98aeb5
Location                                  :
Name                                      : AzMigratePWSHTc8d1replicationfabric
RolloverEncryptionDetailKekCertExpiryDate :
RolloverEncryptionDetailKekCertThumbprint :
RolloverEncryptionDetailKekState          : None
Type                                      : Microsoft.RecoveryServices/vaults/replicationFabrics



PS C:\Users\prsadhu\Documents> $obj | ConvertTo-Json
{
    "BcdrState":  "Valid",
    "CustomDetailInstanceType":  "VMwareV2",
    "EncryptionDetailKekCertExpiryDate":  null,
    "EncryptionDetailKekCertThumbprint":  null,
    "EncryptionDetailKekState":  "None",
    "FriendlyName":  "AzMigratePWSHTc8d1replicationfabric",
    "Health":  "Normal",
    "HealthErrorDetail":  [

                          ],
    "Id":  "/Subscriptions/7c943c1b-5122-4097-90c8-861411bdd574/resourceGroups/azmigratepwshtestasr13072020/providers/Microsoft.RecoveryServices/vaults/AzMigrateTestProjectPWSH02aarsvault/replicationFabrics/AzMigratePWSHTc8d1replicationfabric",
    "InternalIdentifier":  "501ff8f2-c9d7-5bf4-87ff-d0b3fc98aeb5",
    "Location":  null,
    "Name":  "AzMigratePWSHTc8d1replicationfabric",
    "RolloverEncryptionDetailKekCertExpiryDate":  null,
    "RolloverEncryptionDetailKekCertThumbprint":  null,
    "RolloverEncryptionDetailKekState":  "None",
    "Type":  "Microsoft.RecoveryServices/vaults/replicationFabrics"
}